contextily/plotting.py

```diff
@@ -146,6 +148,11 @@ def add_basemap(
     if reset_extent:
         ax.axis((xmin, xmax, ymin, ymax))
 
+    # Add attribution text
+    if url is None:
+        url = providers.Stamen.Terrain
+    if isinstance(url, dict) and attribution is None:
+        attribution = url.get("attribution")
```

Comment:
It's a little bit unfortunate that we need to do the above in more than one place, but I don't think this is avoidable

Reply:
Could we maybe turn it into a method that we can use throughout? Something like:

```python
def check_attribution(url, attribution):
    if url is None:
        url = providers.Stamen.Terrain
    if isinstance(url, dict) and attribution is None:
        attribution = url.get("attribution")
    return url, attribution
```

Reply:
For now it is only in 2 places. So maybe let's factor out once there is a third ;)